### PR TITLE
Two factory auth

### LINKED ISSUE #59: Integrate with Twilio for 2FA
In the system level (initially as an environment variable) one should be able to add Twilio Token for 2FA. If available each user edit form should have a checkbox to enable 2FA. [This tutorial](https://www.twilio.com/blog/2016/01/two-factor-authentication-in-rails-4-with-devise-authy-and-puppies.html) should guide how to implement it.

Important: if the system has no Twilio API configured, everything should work just fine without 2FA. And by adding 2FA, in the activeadmin, the Super Admin should be able to force 2FA to users or disable it if necessary.

### PR DESCRIPTION
**Pull request by rscardinho**
Adding 2FA to users and admins.

Closes #59 

How it works:
An admin can "force" 2FA for an user by setting `Authy enabled` field as true. The setup will be asked on next users login.
An user can also enable 2FA for himself on its profile page. A button on the bottom of the page will ask him to setup the 2FA properly, and a button for disabling 2FA will be displayed. A token verification is required to disable 2FA for users

screen shot 2016-10-20 at 6 47 45 pm
screen shot 2016-10-20 at 6 48 35 pm

### CODE REVIEW

File: config/locales/devise.authy.en.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+en:
```
Comment (daniloisr) [inline config/locales/devise.authy.en.yml]: We will need this file in pt-br too, but I can help you with this if you want :]

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -4,6 +4,9 @@
     <title><%= "#{@project.name} - " if @project %><%= t('cm-central') %></title>
     <%= csrf_meta_tag %>
     <%= stylesheet_link_tag :application %>
+  <%=javascript_include_tag "https://www.authy.com/form.authy.min.js" %>
+  <%=stylesheet_link_tag "https://www.authy.com/form.authy.min.css" %>
```
Comment (daniloisr) [inline app/views/layouts/application.html.erb]: small style problem here: add one more level of indentation and a space after `<%`

File: app/views/devise/devise_authy/verify_authy.html.erb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  <%= verify_authy_form do %>
+    <legend><%= I18n.t('submit_token_title', {:scope => 'devise'}) %></legend>
+    <%= label_tag :token %>
+    <%= text_field_tag :token, "", :autocomplete => :off, :id => 'authy-token' %>
+
+    <!-- Help tooltip -->
+    <!-- You need to configure a help message. -->
+    <!-- See documentation: https://github.com/authy/authy-form-helpers#help-tooltip -->
+    <!-- <%= link_to '?', '#', :id => 'authy-help' %> -->
```
Comment (daniloisr) [inline app/views/devise/devise_authy/verify_authy.html.erb]: I think that you can remove these comments

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (rscardinho) [inline app/views/layouts/application.html.erb]: yeah, `devise-authy` added these lines and I didn't checked this hahah

File: db/schema.rb [hunk 4]
```diff
@@ -50,20 +50,24 @@
...
     t.datetime "current_sign_in_at"
     t.datetime "last_sign_in_at"
     t.inet     "current_sign_in_ip"
     t.inet     "last_sign_in_ip"
-    t.datetime "created_at",                                      null: false
-    t.datetime "updated_at",                                      null: false
+    t.datetime "created_at",                                          null: false
+    t.datetime "updated_at",                                          null: false
+    t.string   "authy_id"
+    t.datetime "last_sign_in_with_authy"
```
Comment (daniloisr) [inline db/schema.rb]: There is no migration that adds authy fields to `admin_users` table, maybe your db/schema.rb is out of sync, right?

Comment (rscardinho) [inline db/schema.rb]: ah, there is no 2FA for admin_users on this PR.
Some conflicts came up between admin_users and users, so I opted to add 2FA for users first, then enable it for admin_users.

File: spec/features/registrations_spec.rb [hunk 5]
```diff
@@ -0,0 +1,83 @@
+require 'feature_helper'
+
+describe "Registrations" do
+  describe 'disable 2FA' do
```
Comment (daniloisr) [inline spec/features/registrations_spec.rb]: Would be nice to have a test where the 2FA is enabled. Do you think that it is easy to add on this PR?

File: app/assets/javascripts/devise_authy.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  $('a#authy-request-sms-link').unbind('ajax:success');
+  $('a#authy-request-sms-link').bind('ajax:success', function(evt, data, status, xhr) {
+    alert(data.message);
+  });
+
+  $('a#authy-request-phone-call-link').unbind('ajax:success');
+  $('a#authy-request-phone-call-link').bind('ajax:success', function(evt, data, status, xhr) {
+    alert(data.message);
+  });
+});
```
Comment (adbatista) [inline app/assets/javascripts/devise_authy.js]: why this file?

Comment (rscardinho) [inline app/assets/javascripts/devise_authy.js]: we're not using it anywhere. deleting

**Pull request merged**